pkg/component/gardener/apiserver/apiserver_test.go

```diff
@@ -1273,6 +1273,55 @@ kubeConfigFile: /etc/kubernetes/admission-kubeconfigs/validatingadmissionwebhook
 				})
 			})
 
+			Context("ManagedResource reconciliation delay", func() {
+				It("should fail if gardener-apiserver service does not get created from ManagedResource", func() {
+					DeferCleanup(test.WithVars(
+						&IntervalWaitServiceGardenerApiserver, time.Millisecond,
+						&TimeoutWaitServiceGardenerApiserver, 5*time.Millisecond,
+					))
+
+					Expect(fakeClient.Delete(ctx, serviceRuntime)).To(Succeed())
+					Eventually(ctx, func(g Gomega) {
+						g.Expect(fakeClient.Get(ctx, client.ObjectKeyFromObject(serviceRuntime), &corev1.Service{})).To(BeNotFoundError())
+					}).Should(Succeed())
+
+					Expect(deployer.Deploy(ctx)).To(MatchError(ContainSubstring("failed waiting for service gardener-apiserver to get created by gardener-resource-manager")))
```

Comment:
Sorry, I lgtm-ed too soon. 
Could you adapt the error messages in the tests to the modified errors in the prod code? Additionally, can you check why the name and namespace of the service is empty in the test output:

```
msg: "failed waiting for service / to get created by gardener-resource-manager: retry failed with max attempts reached, last error: service / was not yet created",
```